library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```
@@ -0,0 +1,592 @@
+  {
+    this.tableName = tableName;
+  }
+
+  public KeyValPair<String, String> getRangeQueryPair()
+  {
+    return rangeQueryPair;
+  }
+
+  public void setRangeQueryPair(KeyValPair<String, String> rangeQueryPair)
+  {
+    this.rangeQueryPair = rangeQueryPair;
+  }
+
+  public int getBatchSize()
+  {
+    return batchSize;
+  }
+
+  public void setBatchSize(int batchSize)
+  {
+    this.batchSize = batchSize;
+  }
+
+  public AbstractJdbcPollInputOperator()
+  {
+    this.pollInterval = 10 * 1000;
+    this.fetchSize = 10000;
+    currentWindowRecoveryState = new MutablePair<>();
+    windowManager = new WindowDataManager.FSWindowDataManager();
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    super.setup(context);
+    spinMillis = context.getValue(Context.OperatorContext.SPIN_MILLIS);
+    execute = true;
+    cause = new AtomicReference<Throwable>();
+    emitQueue = new ArrayBlockingQueue<List<T>>(queueCapacity);
+    this.context = context;
+    operatorId = context.getId();
+
+    try {
+      if (ps == null) {
+        //If its a range query pass upper and lower bounds
+        //If its a polling query pass only the lower bound
+        if (getRangeQueryPair().getValue() != null) {
+          ps = store.getConnection().prepareStatement(
+              JdbcMetaDataUtility.buildRangeQuery(getTableName(), getKey(), rangeQueryPair.getKey(),
+                  rangeQueryPair.getValue()),
+              java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+        } else {
+          ps = store.getConnection().prepareStatement(
+              JdbcMetaDataUtility.buildPollableQuery(getTableName(), getKey(), rangeQueryPair.getKey()),
+              java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+          isPollable = true;
+        }
+      }
+    } catch (SQLException e) {
+      LOG.error("Exception in initializing the range query for a given partition", e);
```

why swallow exception? Throw exception